com.unity.render-pipelines.universal/Editor/ShaderGUI/BaseShaderGUI.cs

```diff
@@ -325,19 +404,53 @@ protected static void DrawTileOffset(MaterialEditor materialEditor, MaterialProp
         ////////////////////////////////////
         #region MaterialDataFunctions
 
-        public static void SetMaterialKeywords(Material material, Action<Material> shadingModelFunc = null, Action<Material> shaderFunc = null)
+        // this function is shared with ShaderGraph Lit/Unlit GUIs and also the hand-written GUIs
+        internal static void UpdateMaterialSurfaceOptions(Material material)
         {
             // Setup blending - consistent across all Universal RP shaders
             SetupMaterialBlendMode(material);
 
+            bool isShaderGraph = material.IsShaderGraph();
+
+            // Cast Shadows
+            bool castShadows = true;
+            if (material.HasProperty(Property.CastShadows))
+            {
+                castShadows = (material.GetFloat(Property.CastShadows) != 0.0f);
+            }
+            else
+            {
+                if (isShaderGraph)
+                {
+                    // Lit.shadergraph or Unlit.shadergraph, but no material control defined
+                    // enable the pass in the material, so shader can decide...
+                    castShadows = true;
+                }
+                else
+                {
+                    // Lit.shader or Unlit.shader -- set based on transparency
+                    castShadows = Rendering.Universal.ShaderGUI.LitGUI.IsOpaque(material);
+                }
+            }
+            material.SetShaderPassEnabled("ShadowCaster", castShadows);
+
             // Receive Shadows
-            if (material.HasProperty("_ReceiveShadows"))
-                CoreUtils.SetKeyword(material, "_RECEIVE_SHADOWS_OFF", material.GetFloat("_ReceiveShadows") == 0.0f);
+            if (material.HasProperty(Property.ReceiveShadows))
+                CoreUtils.SetKeyword(material, ShaderKeywordStrings._RECEIVE_SHADOWS_OFF, material.GetFloat(Property.ReceiveShadows) == 0.0f);
+        }
 
-            // Setup double sided GI
-            bool doubleSidedGI = (RenderFace)material.GetFloat("_Cull")  != RenderFace.Front;
-            if (doubleSidedGI != material.doubleSidedGI)
-                material.doubleSidedGI = doubleSidedGI;
+        // this is the function used by Lit.shader, Unlit.shader GUIs
+        public static void SetMaterialKeywords(Material material, Action<Material> shadingModelFunc = null, Action<Material> shaderFunc = null)
```

Comment:
Does it need to be public/protected? If so, needs public API docs.

Reply:
The function signature is unchanged in this PR (just the diff is bad here -- it's not matching up correctly with the function declaration -- should be line ~328 in original)

Reply:
changed it to internal